### PR TITLE
Update SDKs for google provider package

### LINKED ISSUE #27292: Support for Python 3.11 for Google Provider (upgrading all dependencies)
I know it is eaarly (Python 3.11 has just been released yesterday) but we are hoping in Apache Airflow to a much faster cycle of adding new Python releases - especially that Pyhon 3.11 introduces huge performance improvements (25% is the average number claimed) due to a very focused effort to increase single-threaded Python performance (Specialized interpreter being the core of it but also many other improvements) without actually changing any of the Python code.

The google provider will be a huge drag on Airlfow's compatibility for Python 3.11 and we might even decide to release Airlfow without Google Provider support for 3.11. Though it would be great to avoid that.

Google Provider (as it was originally mentioned in https://github.com/apache/airflow/issues/12116) has still a number of old google cloud libraries < 2.0.0 that for sure will not get good. Also support for libraries such as bigquery for Python 3.11 has to be added (but this is external to it and tracked in  https://github.com/googleapis/python-bigquery/issues/1386) 

Nice summary of Py3.11 support is here: https://pyreadiness.org/3.11/ - it's not very green obviously, but I hope it gets greener soon.

I just opened the PR to add 3.11 support yesterday and plan to keep it open until it gets green :)

https://github.com/apache/airflow/pull/27264

I think it would be fantastic if we could work out all the problems and add migrate all the old dependencies:

- [ ] google-api-python-client>=1.6.0,<2.0.0
- [x] google-cloud-bigtable>=1.0.0,<2.0.0 https://github.com/apache/airflow/pull/29644
- [x] google-cloud-dlp>=0.11.0,<2.0.0 https://github.com/apache/airflow/pull/29234
- [ ] google-cloud-language>=1.1.1,<2.0.0
- [ ] google-cloud-secret-manager>=0.2.0,<2.0.0
- [ ] google-cloud-spanner>=1.10.0,<2.0.0
- [ ] google-cloud-speech>=0.36.3,<2.0.0
- [ ] google-cloud-texttospeech>=0.4.0,<2.0.0
- [ ] google-cloud-translate>=1.5.0,<2.0.0  
- [ ] google-cloud-videointelligence>=1.7.0,<2.0.0
- [ ] google-cloud-vision>=0.35.2,<2.0.0
- [x] python-bigquery-sqlalchemy is a blocker for PROD images:  https://github.com/googleapis/python-bigquery-sqlalchemy/issues/500

Looking forward to cooperation on that one :)

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by lwyszomi**
As everyone know google provider package have a lot of old dependencies. I would like to start migration to the latest versions of the SDK. For now we are blocked by some other dependencies because they are using `protobuf<4`.

```
apache-beam
mysql-connector-python
yandexcloud
```

Also in the google SDKs we had a lot of breaking changes so after updating we need to adjust broken operators. I did investigation how big is this problem and I'm attaching the list of services where some of the operators are broken:

- [x] AutoML -> need investigation
- [x] BigQuery -> need investigation
- [x] BigTable -> need investigation
- [x] CloudBuild -> need investigation
- [x] CloudFunctions -> need investigation
- [x] CloudMemorystore -> OK
- [x] CloudSQL -> need investigation
- [x] Composer -> adjust system tests
- [x] Compute -> need adjustments in system tests
- [x] DataLossPrevention -> OK
- [x] Dataflow -> need investigation
- [x] dataform -> OK
- [x] datafusion -> OK
- [x] dataplex -> need investigation
- [x] dataprep -> need investigation
- [x] Dataproc -> need investigation
- [x] dataprocMetastore -> need investigation
- [x] datastore -> need adjustments in system tests
- [x] GCS -> need investigations
- [x] kubernetesEngine -> OK
- [x] LifeSciences -> OK
- [x] MLEngine -> need investigation
- [x] NaturalLanguage -> need investigation
- [x] PubSub - OK
- [x] Spanner - OK
- [x] SpeachToTet -> need investigation
- [x] SQLToSheets -> need investigation
- [x] StackDriver -> need investigation 
- [x] StorageTransfer -> need investigation
- [x] Tasks -> need investigation
- [x] TextToSpeech -> need investigation
- [x] Transfers -> Need investigation
- [x] Translate -> OK
- [x] TranslateSpeech -> Need investigation
- [x] VertexAI - Need Investigation
- [x] VideoIntelligence - need investigation
- [x] Vision - need investigation
- [x] Workflows - OK

Fixes: #27292

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by felicienveldema** [comment]
Got referred to here from https://github.com/apache/airflow/issues/27292 .
I'm experiencing an issue where I am not able to upgrade the deprecated Google Ads API 18. 

I eventually get stuck with apache-airflow-providers-google depending on google-cloud-secret-manager < 2.x . Which depend on protobuf 3 which causes my predicament.  Higher versions depend on protobuf > 4.5.x .

Is there progress on this ticket? Just wondering but keep up the good workd

**Comment by lwyszomi** [comment]
@felicienveldema We are still working on the changes in the google-provider package, but still we have some problems with the dependencies in 3 packages because they are still depends on protobuf<4.0 and probably at May we will have updates. So I think the will have new google-provider version supporting latest version of SDKs at the end of May or later.

**Comment by r-richmond** [comment]
@potiuk

I got curious about the 3 listed packages above causing issues. One of them `mysql-connector-python`'s latest version has `protobuf` [pinned at 3.20.3](https://github.com/mysql/mysql-connector-python/blob/735322dbe0b453b01d3e52392cf364717b78bdb8/setup.py#L142). How do you anticipate this conflict will be resolved? (After reviewing the commit activity & Oracle's ownership I'm assuming they will be slow to update if they do at all, apologies if this is a bad assumption).

**Comment by potiuk** [comment]
> I got curious about the 3 listed packages above causing issues. One of them `mysql-connector-python`'s latest version has `protobuf` [pinned at 3.20.3](https://github.com/mysql/mysql-connector-python/blob/735322dbe0b453b01d3e52392cf364717b78bdb8/setup.py#L142). How do you anticipate this conflict will be resolved? (After reviewing the commit activity & Oracle's ownership I'm assuming they will be slow to update if they do at all, apologies if this is a bad assumption).

I have not looked at it yet. Do you have some ideas ?

**Comment by potiuk** [comment]
Generally the options are:

* replace the library with something else
* exclude such provider that holds us back
* vendor-in the requuirement and bump the dependency
* make the dependency optional and skip tests for it

So we have a number of options we can follow.

**Comment by potiuk** [comment]
I just opened the issue to: yandex-cloud:  https://github.com/yandex-cloud/python-sdk/issues/71 and I will prepare support for disabling providers and excluding them if they are holding us back (cc: @eladkal). 

I will also raise this to our devlist.

**Comment by potiuk** [comment]
Devlist discussion started: https://lists.apache.org/thread/j98bgw9jo7xr4fvjh27d6bfoyxr1omcm (especially CC: @eladkal especially) I am curious what you think.

**Comment by potiuk** [comment]
FYI: We have no problem with apache-beam: https://github.com/apache/beam/pull/24599 - 2 weeks ago they marged protobuf bump, so we just neeed to wait for the next release

**Comment by potiuk** [comment]
We have first reaction: https://github.com/yandex-cloud/python-sdk/issues/71#issuecomment-1484734024

**Comment by r-richmond** [comment]
>But I am also all for disabling mysql provider if they don't respond.

I'm also 100% for this (but I've got some bias in that I always use Postgres over mysql). I hesitated to suggest that since I was unsure if that would impact offering mysql as one of Airflow Meta DB backends.

**Comment by potiuk** [comment]
> I'm also 100% for this (but I've got some bias in that I always use Postgres over mysql). I hesitated to suggest that since I was unsure if that would impact offering mysql as one of Airflow Meta DB backends.

I have to check but I think this has actually nothing to do with mysql metadata backend. For that we are using sqlalchemy and it has a few drivers it can choose from. And I think our driver for CI/tests is `mysqlclient` not mysql-connector-python. 

BTW. This is another possibilty to rewrite the hooks to use mysqlclient. I might take a look at that actually.

**Comment by cgadam** [comment]
> Hi, is it too risky for Airflow to just update from `google-ads` `v18.0.0` to `v18.2.0`? See: #30353
> 
> Today `v11` of Google API is sunsetting: https://developers.google.com/google-ads/api/docs/sunset-dates which means that current latest version of Airflow won't be officially compatible (due to its constraint file: https://raw.githubusercontent.com/apache/airflow/constraints-2.5.2/constraints-3.7.txt) with any `google-ads` package that can actually interact with the Google Ads API. (API calls will start failing)
> 
> Latest compatibility to a new API version was added in: https://github.com/googleads/google-ads-python/pull/672/files#diff-91c5b46dc84a94604a4e4d0caed9bf85590a2eddbb12d2e8dc80badf324a9dfbR9 (`v17.0.0`) and it added support `v11` of the API.
> 
> `v18.2.0` actually added support for `v12` of the API. See [here](https://github.com/googleads/google-ads-python/pull/716/files#diff-91c5b46dc84a94604a4e4d0caed9bf85590a2eddbb12d2e8dc80badf324a9dfbR1).

We're in the dark night now. Sunset has passed 😅 We're not getting error: `"Version v11 is deprecated. Requests to this version will be blocked."`

**Comment by potiuk** [comment]
> I'm wondering, because after removing the `mysql-connector-python` this feature will be no longer needed and can be removed as well. However, new libraries might appear in the future, and we will probably need it back then, so in this case it would be nice to save it for future use even if there will be only one option available. WDYT @potiuk, should we save it?

I have not thought about it yet. I am waiting for the response of Oracle (If it comes) for a week - according to our new policy that's being "lazy consent now" and then I will take a closer look at that. There is also an option to tunr mysql-connector-python into **ACTUALLY** optional feature (which I think is the best option) - so make it an extra (we already have a few of those). In this case we should leave it.

**Comment by potiuk** [comment]
@cgadam: It is likely we **might** have a proposal how to solve it soon - would you be willing to test it if I give you access to a beta/pre-release of google provider that you could test with it with an implemented worakround (with an intention of making it into next release?)

**Comment by potiuk** [comment]
> I'm wondering, because after removing the mysql-connector-python this feature will be no longer needed and can be removed as well. However, new libraries might appear in the future, and we will probably need it back then, so in this case it would be nice to save it for future use even if there will be only one option available. WDYT @potiuk, should we save it?

* Re that : #30487 has been implemented wth making the connector optional
* With yandexcloud : #30667  is on it's way to get lazy consensus by Wednesday
* Apache Beam is gearing for not only Protobuf 4 but also Python 3.11 release mid May

I think we are on a good track to get there. BTW. If you would like to test a PR with Google-only upgrades @lwyszomi  and the rest of the team, you should be able to create a PR with "Suspended: true" for the providers that are affecting you (yandexcloud and beam currently) and run yout tests there.

**Comment by potiuk** [comment]
Looks like the last missing bit is apache-beam. They released 2.47.0rc1 yesterday with both pyarrow and protobuf updated. I pushed a fixup for now fixing it to that version (including regenerated `generated/provider_dependencies.json` - let's see where it gets at.

**Comment by potiuk** [comment]
cc: @lwyszomi

File: airflow/providers/apache/beam/provider.yaml
Comment (r-richmond) [inline airflow/providers/apache/beam/provider.yaml]:
```suggestion
  - apache-beam>=2.47.0rc1
```
@potiuk Should it be this instead?

Comment (potiuk) [reply airflow/providers/apache/beam/provider.yaml]: Finally it should be likely >= 2.47 when they release. For now - I wanted to make sure it works, and put ==  to unblock @lwyszomi and the team to start solving test failures as the image could not be built without it due to conflicts.

And we are in a good place finally. The image builds without conflicts :tada: !!! and those are the upgraded packages  -- exactly what we wanted :) 

```
< # This constraints file was automatically generated on 2023-04-27T21:53:27Z
---
> # This constraints file was automatically generated on 2023-04-27T23:10:05Z
43a44
> Shapely==1.8.5.post1
63c64
< apache-beam==2.46.0
---
> apache-beam==2.47.0rc1
108c109
< black==23.1a1
---
> black==23.3.0
162c163
< elasticsearch-dbapi==0.2.10
---
> elasticsearch-dbapi==0.2.2
164c165
< elasticsearch==7.13.4
---
> elasticsearch==7.17.9
189,190c190,192
< google-api-core==2.8.2
< google-api-python-client==1.12.11
---
> google-ads==21.0.0
> google-api-core==2.11.0
> google-api-python-client==2.86.0
192c194
< google-auth-oauthlib==0.8.0
---
> google-auth-oauthlib==1.0.0
194,204c196,206
< google-cloud-aiplatform==1.16.1
< google-cloud-appengine-logging==1.1.3
< google-cloud-audit-log==0.2.4
< google-cloud-automl==2.8.0
< google-cloud-bigquery-datatransfer==3.7.0
< google-cloud-bigquery-storage==2.14.1
< google-cloud-bigquery==2.34.4
< google-cloud-bigtable==2.11.1
< google-cloud-build==3.9.0
< google-cloud-compute==0.7.0
< google-cloud-container==2.11.1
---
> google-cloud-aiplatform==1.24.1
> google-cloud-appengine-logging==1.3.0
> google-cloud-audit-log==0.2.5
> google-cloud-automl==2.11.1
> google-cloud-bigquery-datatransfer==3.11.1
> google-cloud-bigquery-storage==2.19.1
> google-cloud-bigquery==3.10.0
> google-cloud-bigtable==2.17.0
> google-cloud-build==3.16.0
> google-cloud-compute==1.11.0
> google-cloud-container==2.21.0
206,225c208,227
< google-cloud-datacatalog==3.9.0
< google-cloud-dataflow-client==0.5.4
< google-cloud-dataform==0.2.0
< google-cloud-dataplex==1.1.0
< google-cloud-dataproc-metastore==1.6.0
< google-cloud-dataproc==5.0.0
< google-cloud-dlp==3.8.0
< google-cloud-kms==2.12.0
< google-cloud-language==1.3.2
< google-cloud-logging==3.2.1
< google-cloud-memcache==1.4.1
< google-cloud-monitoring==2.11.0
< google-cloud-orchestration-airflow==1.4.1
< google-cloud-os-login==2.7.1
< google-cloud-pubsub==2.13.5
< google-cloud-redis==2.9.0
< google-cloud-resource-manager==1.6.0
< google-cloud-secret-manager==1.0.2
< google-cloud-spanner==1.19.3
< google-cloud-speech==1.3.4
---
> google-cloud-datacatalog==3.12.0
> google-cloud-dataflow-client==0.8.3
> google-cloud-dataform==0.5.1
> google-cloud-dataplex==1.4.3
> google-cloud-dataproc-metastore==1.11.0
> google-cloud-dataproc==5.4.1
> google-cloud-dlp==3.12.1
> google-cloud-kms==2.16.1
> google-cloud-language==2.9.1
> google-cloud-logging==3.5.0
> google-cloud-memcache==1.7.1
> google-cloud-monitoring==2.14.2
> google-cloud-orchestration-airflow==1.7.1
> google-cloud-os-login==2.9.1
> google-cloud-pubsub==2.16.0
> google-cloud-redis==2.12.1
> google-cloud-resource-manager==1.10.0
> google-cloud-secret-manager==2.16.1
> google-cloud-spanner==3.32.0
> google-cloud-speech==2.19.0
227,232c229,234
< google-cloud-tasks==2.10.1
< google-cloud-texttospeech==1.0.3
< google-cloud-translate==1.7.2
< google-cloud-videointelligence==1.16.3
< google-cloud-vision==1.0.2
< google-cloud-workflows==1.7.1
---
> google-cloud-tasks==2.13.1
> google-cloud-texttospeech==2.14.1
> google-cloud-translate==3.11.1
> google-cloud-videointelligence==2.11.1
> google-cloud-vision==3.4.1
> google-cloud-workflows==1.10.1
235c237
< googleapis-common-protos==1.56.4
---
> googleapis-common-protos==1.59.0
239c241
< grpc-google-iam-v1==0.12.4
---
> grpc-google-iam-v1==0.12.6
241c243
< grpcio-status==1.48.2
---
> grpcio-status==1.54.0
356,357c358,359
< packaging==21.3
< pandas-gbq==0.17.9
---
> packaging==23.1
> pandas-gbq==0.19.1
385,386c387,388
< proto-plus==1.19.6
< protobuf==3.20.0
---
> proto-plus==1.22.2
> protobuf==4.22.3
394c396
< pyarrow==9.0.0
---
> pyarrow==11.0.0
512c514
< tableauserverclient==0.24
---
> tableauserverclient==0.25
558c560
< uritemplate==3.0.1
---
> uritemplate==4.1.1
```

**Comment by potiuk** [comment]
Seems now there are some test collection problems to solve @lwyszomi  -> and I'd strongly recommend to rebase to the latest main as well, this will speed up the builds ~ 40%/50% times following the recent CI speed-ups I did and will make iteration on CI issues to solve quite a bit faster. And == 2.47rc1 on beam, we can start solving the problems even before they release it so that we can merge and release google provider **right** after `apache.beam` does.

**Comment by potiuk** [comment]
Still some tests/static changes to go - and few tests failing - but that one looks really great:

```
43a44
> Shapely==1.8.5.post1
63c64
< apache-beam==2.46.0
---
> apache-beam==2.47.0
108c109
< black==23.1a1
---
> black==23.3.0
163c164
< elasticsearch-dbapi==0.2.10
---
> elasticsearch-dbapi==0.2.2
165c166
< elasticsearch==7.13.4
---
> elasticsearch==7.17.9
190,191c191,193
< google-api-core==2.8.2
< google-api-python-client==1.12.11
---
> google-ads==21.1.0
> google-api-core==2.11.0
> google-api-python-client==2.86.0
193c195
< google-auth-oauthlib==0.8.0
---
> google-auth-oauthlib==1.0.0
195,205c197,207
< google-cloud-aiplatform==1.16.1
< google-cloud-appengine-logging==1.1.3
< google-cloud-audit-log==0.2.4
< google-cloud-automl==2.8.0
< google-cloud-bigquery-datatransfer==3.7.0
< google-cloud-bigquery-storage==2.14.1
< google-cloud-bigquery==2.34.4
< google-cloud-bigtable==2.11.1
< google-cloud-build==3.9.0
< google-cloud-compute==0.7.0
< google-cloud-container==2.11.1
---
> google-cloud-aiplatform==1.25.0
> google-cloud-appengine-logging==1.3.0
> google-cloud-audit-log==0.2.5
> google-cloud-automl==2.11.1
> google-cloud-bigquery-datatransfer==3.11.1
> google-cloud-bigquery-storage==2.19.1
> google-cloud-bigquery==3.10.0
> google-cloud-bigtable==2.18.0
> google-cloud-build==3.16.0
> google-cloud-compute==1.11.0
> google-cloud-container==2.21.0
207,226c209,228
< google-cloud-datacatalog==3.9.0
< google-cloud-dataflow-client==0.5.4
< google-cloud-dataform==0.2.0
< google-cloud-dataplex==1.1.0
< google-cloud-dataproc-metastore==1.6.0
< google-cloud-dataproc==5.0.0
< google-cloud-dlp==3.8.0
< google-cloud-kms==2.12.0
< google-cloud-language==1.3.2
< google-cloud-logging==3.2.1
< google-cloud-memcache==1.4.1
< google-cloud-monitoring==2.11.0
< google-cloud-orchestration-airflow==1.4.1
< google-cloud-os-login==2.7.1
< google-cloud-pubsub==2.13.5
< google-cloud-redis==2.9.0
< google-cloud-resource-manager==1.6.0
< google-cloud-secret-manager==1.0.2
< google-cloud-spanner==1.19.3
< google-cloud-speech==1.3.4
---
> google-cloud-datacatalog==3.12.0
> google-cloud-dataflow-client==0.8.3
> google-cloud-dataform==0.5.1
> google-cloud-dataplex==1.4.3
> google-cloud-dataproc-metastore==1.11.0
> google-cloud-dataproc==5.4.1
> google-cloud-dlp==3.12.1
> google-cloud-kms==2.16.1
> google-cloud-language==2.9.1
> google-cloud-logging==3.5.0
> google-cloud-memcache==1.7.1
> google-cloud-monitoring==2.14.2
> google-cloud-orchestration-airflow==1.7.1
> google-cloud-os-login==2.9.1
> google-cloud-pubsub==2.16.1
> google-cloud-redis==2.12.1
> google-cloud-resource-manager==1.10.0
> google-cloud-secret-manager==2.16.1
> google-cloud-spanner==3.33.0
> google-cloud-speech==2.19.0
228,233c230,235
< google-cloud-tasks==2.10.1
< google-cloud-texttospeech==1.0.3
< google-cloud-translate==1.7.2
< google-cloud-videointelligence==1.16.3
< google-cloud-vision==1.0.2
< google-cloud-workflows==1.7.1
---
> google-cloud-tasks==2.13.1
> google-cloud-texttospeech==2.14.1
> google-cloud-translate==3.11.1
> google-cloud-videointelligence==2.11.1
> google-cloud-vision==3.4.1
> google-cloud-workflows==1.10.1
236c238
< googleapis-common-protos==1.56.4
---
> googleapis-common-protos==1.59.0
240c242
< grpc-google-iam-v1==0.12.4
---
> grpc-google-iam-v1==0.12.6
242c244
< grpcio-status==1.48.2
---
> grpcio-status==1.54.0
334c336
< nodeenv==1.7.0
---
> nodeenv==1.8.0
358,359c360,361
< packaging==21.3
< pandas-gbq==0.17.9
---
> packaging==23.1
> pandas-gbq==0.19.2
387,388c389,390
< proto-plus==1.19.6
< protobuf==3.20.0
---
> proto-plus==1.22.2
> protobuf==4.22.5
396c398
< pyarrow==9.0.0
---
> pyarrow==11.0.0
412c414
< pymongo==3.13.0
---
> pymongo==4.3.3
514c516
< tableauserverclient==0.24
---
> tableauserverclient==0.25
559c561
< uritemplate==3.0.1
---
> uritemplate==4.1.1
```

**Comment by potiuk** [comment]
The "React" failures seem to be unrelated - some  intermittent problem with yarn package we use, we can re-run it after all others complete.

**Comment by potiuk** [comment]
Also to note - it 
Fixes: #27292

File: airflow/providers/google/provider.yaml
Comment (kosteev) [inline airflow/providers/google/provider.yaml:121]: Can you, please, put it in lexicographical order

**Review by kosteev — APPROVED** [review]

**Comment by r-richmond** [comment]
> Also to note - it
> Fixes: #27292

It makes great progress towards that ticket but it does not resolve it 100%. As noted [here](https://github.com/apache/airflow/issues/27292#issuecomment-1454832422) one of the google libraries doesn't support 3.11 yet (and they still haven't implemented that yet https://github.com/googleapis/python-aiplatform/pull/2028)

**Comment by potiuk** [comment]
> > Also to note - it
> > Fixes: #27292
> 
> It makes great progress towards that ticket but it does not resolve it 100%. As noted [here](https://github.com/apache/airflow/issues/27292#issuecomment-1454832422) one of the google libraries doesn't support 3.11 yet (and they still haven't implemented that yet [googleapis/python-aiplatform#2028](https://github.com/googleapis/python-aiplatform/pull/2028))

Thanks for that info!  yeah I see where it is not 100% but  I think it this case the 99% is "good enough" to close it and open another issue for the "python-api" exclusively.  Before that we really could not even think about releasing 3.11 (and especially with the underlying problem with protobuf, apaache-beam and depending libraries). But I am perfectly happy to add `python-aiplatform` dependency to be excluded for Python >= 3.11 for example.... 

So yeah. I think it really "fixes" the problem :D as I see it.

**Comment by potiuk** [comment]
👀 👀 👀 👀

**Comment by potiuk** [comment]
Those are intermittent errors only (I need to make them more stable). Merging

**Review by potiuk — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 
🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 
🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉 🎉

**Comment by potiuk** [comment]
CC: @ephraimbuddy -> I just realized we will need it - I marked this one also for 2.6.2. While the "code" changes aren't used in the release from 2.6.2, the "dependency" part (providers and generated/provider_dependencies.json) will be needed to properly build CI once we release the new google provider with all its deps

**Comment by eladkal** [comment]
> CC: @ephraimbuddy -> I just realized we will need it - I marked this one also for 2.6.2. While the "code" changes aren't used in the release from 2.6.2, the "dependency" part (provider.yaml and generated/provider_dependencies.json) will be needed to properly build CI once we release the new google provider with all its deps

I'm hoping that we will release 2.6.2 as followup right after provider wave is released.

**Pull request merged**
